README.md

```diff
@@ -164,6 +164,13 @@ the following fields:
   implementations that restart the system as part of their normal operation
   and expect to start up with the same config after a restart.
 
+  **WARNING**: in environments where tasks runs on the host operating system,
+  this functionality can expose the worker's credentials to the task.  The
+  process creates this configuration file with read permission only for the
+  current user. As an added layer of protection, we recommend using a filename
+  located in a directory that is only accessible by the user running
+  `start-worker`. 
+
```

Comment:
Note, `generic-worker` actively takes ownership of the config file on startup, making it unreadable to task users, so this isn't a concern for generic-worker. On Windows, the `os` package functions mostly ignore the file permission flags that are passed in. Instead, explicit ACLs typically need to be created, but this is handled by generic-worker.